### PR TITLE
Add property for OLED display sleep mode

### LINKED ISSUE #24: Feature request: display sleep/wake functions
I was trying to sleep/wake the device (to save the shiny parts) when not in use. My usual technique (on TFTs anyway) is to set `display.brightness = 0` to turn off the backlight, but I discovered that the command told by the source (in circuitpython core) for this board for brightness is actually "contrast" according to the controller datasheet (oled -> no backlight), which explains why it got dimmer but did not turn off completely.

The standalone SH1107 driver has `sleep()` and `wake()` functions that use explicit sleep/wake commands: https://github.com/adafruit/Adafruit_CircuitPython_DisplayIO_SH1107/blob/main/adafruit_displayio_sh1107.py#L164  but this board's display support is built-in using C, and no such functions are there. I ended up just mangling a copy of that library to wrap an existing displayio display, which works, but is not awesome. (see https://gist.github.com/rpavlik/73d1de102038f85de051fc1b8cdd3009 )

Not sure if "sleep" and "wake" can be monkey-patched in from Python into the C displayio class, or if there's some other, better way to provide those functions, but I figured this would be the library to do them in.

### PR DESCRIPTION
**Pull request by kalehmann**
I am interested in putting the OLED display of the MacroPad to sleep after a longer period of inactivity. The issue #24 already lines out how the display is put to sleep and mentions, that this functionality could be added to the MacroPad library.

This pull request adds a property `display_sleep` to the class `MacroPad`. Changing the property puts the OLED display to sleep or wakes it up after sleep.

This closes #24

### CODE REVIEW

File: adafruit_macropad.py [hunk 1]
```diff
@@ -324,6 +324,25 @@ def _keys_and_pixels(
...
+        return self._display_sleep
+
+    @display_sleep.setter
+    def display_sleep(self, sleep: bool) -> None:
+        """Change the sleep status of the display.
+
+        If the display is put to sleep, it stops the OLED drive and greatly
+        reduces its power usage. The current content of the display is kept
+        in the memory and can still be updated.
+        """
```
Comment (Neradoc) [inline adafruit_macropad.py]: the docstring in the setter is ignored, it should all be in the getter docstring (in the docs, a `@property` is a single entry). Changing the first line maybe to "The current sleep status of the display, set it to put the display to sleep or wake it." etc..

```suggestion
    def display_sleep(self) -> bool:
        """
        The current sleep status of the display, set it to put the display to sleep or wake it.
        
        If the display is put to sleep, it stops the OLED drive and greatly
        reduces its power usage. The current content of the display is kept
        in the memory and can still be updated.
        """
        return self._display_sleep

    @display_sleep.setter
    def display_sleep(self, sleep: bool) -> None:
        """Change the sleep status of the display."""
```

**Review by Neradoc — CHANGES_REQUESTED** [review]
Oh I was doing it without seeing that you already did !
A little note: the docstring should be all in the getter.
Also I don't think that the use of the ternary operator helps here, I'd prefer a more explicit if/else, but maybe that's just me.
Thanks for the PR !

File: adafruit_macropad.py (hunk 1 above)
Comment (kalehmann) [reply adafruit_macropad.py]: Thank you for the review. I didn't even know, that the docstring for the setter is ignored by sphinx.

I put the whole documentation into the setter now, added some more details from the datasheet and replaced the ternary.

**Review by Neradoc — APPROVED** [review]
Thank you !

**Pull request merged**